src/main/java/dal/fatih/todoly/TaskRepository.java

```diff
@@ -1,165 +1,23 @@
-    private final PreparedStatement deletePreparedStatement;
-    private final PreparedStatement filterPreparedStatement;
-    private final PreparedStatement getByTitleOrDesPreparedStatement;
-
-
-    public TaskRepository() throws SQLException {
-        createTable();
-        createPrepareStatement =
-                connection.prepareStatement("insert into tasks " + "values (?,?,?,?,?)");
-        listPrepareStatement =
-                connection.prepareStatement("select id,taskid,title,description,duedate from tasks");
-        getPreparedStatement =
-                connection.prepareStatement("select taskid,title,description,duedate from tasks where taskid=? limit 1");
-        deletePreparedStatement =
-                connection.prepareStatement("delete from tasks where taskid=?");
-        filterPreparedStatement =
-                connection.prepareStatement("select * from tasks where duedate between now() and ?");
-        getByTitleOrDesPreparedStatement =
-                connection.prepareStatement("select taskid,title,description,duedate from tasks where title like ? "
-                        + "or description like ?");
-    }
-
-    public void createTable() {
-        try {
-            Statement statement = connection.createStatement();
-            statement.executeUpdate("create table if not exists tasks"
-                    + "  (id int auto_increment primary key not null,"
-                    + "   taskid uuid(36),"
-                    + "   title varchar(35),"
-                    + "   description varchar(250),"
-                    + "   duedate date)");
-            statement.close();
-        } catch (SQLException e) {
-            throw new RuntimeException(e);
-        }
-    }
-
-    public boolean create(Task task) {
-        try {
-            createPrepareStatement.setString(1, null);
-            createPrepareStatement.setString(2, task.getId().toString());
-            createPrepareStatement.setString(3, task.getTitle());
-            createPrepareStatement.setString(4, task.getDescription());
-            createPrepareStatement.setString(5, task.getDueDate().toString());
-            createPrepareStatement.executeUpdate();
-            return true;
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
-    }
-
-    public List<Task> list() {
-        List<Task> tasks = new ArrayList<>();
-        try {
-            ResultSet resultSet = listPrepareStatement.executeQuery();
-            while (resultSet.next()) {
-                Task task = new Task();
-                task.setId(resultSet.getObject(2, java.util.UUID.class));
-                task.setTitle(resultSet.getString(3));
-                tasks.add(task);
-            }
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
-        return tasks;
-    }
+public interface TaskRepository {
 
-    public Task get(String taskIdInput) {
+    void createTable();

```

Comment:
Don't make this an interface method since it is only for JDBC one and not the hibernate one.